environs/tools/validation.go

```diff
@@ -55,9 +51,12 @@ func ValidateToolsMetadata(params *ToolsMetadataLookupParams) ([]string, *simple
 			},
 			Stream: params.Stream,
 			Series: []string{params.Series},
-			Arches: params.Architectures,
 		})
 	}
+	if len(params.Architectures) != 0 {
```

Comment:
do we need to make this conditional? is a nil slice treated differently to a non-nil but empty slice?